### PR TITLE
Add example for klog v1 project using klog v2 library

### PR DESCRIPTION
**Pull request by dims**
Examples on how to use v2 based library in a primarily klog v1 project (how to plug gaps until full migration to v2)

### CODE REVIEW

File: klog.go [hunk 1]
```diff
@@ -831,6 +831,7 @@ func SetLogger(logr logr.Logger) {
 func SetOutput(w io.Writer) {
 	logging.mu.Lock()
 	defer logging.mu.Unlock()
+	logging.toStderr = false
```
Comment (dims) [inline klog.go:834]: SetOutput implies, nothing should go to stderr !!

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+type klogWriter struct {}
+func (kw klogWriter)Write(p []byte) (n int, err error){
+	klog.Info(string(p))
+	return len(p), nil
+}
+
+func main() {
+	klog.InitFlags(nil)
+	klogv2.SetOutput(klogWriter{})
```
Comment (dims) [inline examples/coexist_klog_v1/coexist_klog_v1.go:18]: This example uses `SetOutput`

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func (kw klogLogger) Write(p []byte) (n int, err error) {
+	klog.Info(string(p))
+	return len(p), nil
+}
+
+func main() {
+	klog.InitFlags(nil)
+	klogv2.SetLogger(klogLogger{})
```
Comment (dims) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:44]: This one uses `SetLogger`

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"flag"
+
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+type klogWriter struct {}
+func (kw klogWriter)Write(p []byte) (n int, err error){
```
Comment (neolit123) [inline examples/coexist_klog_v1/coexist_klog_v1.go:11]: there should be a space before `Write`.
the github actions / gofmt should be catching this, but it doesn't seem like so.

Comment (neolit123) [reply examples/coexist_klog_v1/coexist_klog_v1.go:11]: i don't know how this linter works, but it seems we are telling it to do gofmt:
https://github.com/kubernetes/klog/blob/master/.github/workflows/test.yml#L32

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	klog.Info(string(p))
```
Comment (liggitt) [inline examples/coexist_klog_v1/coexist_klog_v1.go:12]: doesn't this misattribute line numbers, mess up vmodule, and lose info/warn/fatal distinctions?

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+func (kw klogLogger) Info(msg string, keysAndValues ...interface{}) {
+	klog.Info(msg, keysAndValues)
+}
+
+func (kw klogLogger) Enabled() bool {
+	return true
+}
+
+func (kw klogLogger) Error(err error, msg string, keysAndValues ...interface{}) {
+	klog.Error(err, msg, keysAndValues)
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:22]: same question about line numbers, vmodule, and klog2.Fatalf behavior

File: klog.go (hunk 1 above)
Comment (liggitt) [reply klog.go:834]: is this a breaking change?

Comment (dims) [reply klog.go:834]: more like unspecified behavior i think. example how folks have work around when all they want to do is discard the logs ...
https://github.com/cri-o/cri-o/blob/master/cmd/crio/main.go#L103-L110

**Comment by 44past4** [comment]
Wouldn't having a bridge implementation of klog v1 interface on top of klog v2 similar to the one below help with the migration? With such implementation to use both v1 and v2 in the same project it would be sufficient to simply add v2 implementation to the project and then replace v1 with the bridge implementation. 

```go
package klog

import (
	klogv2 "k8s.io/klog/v2"
)

type Level = klogv2.Level

// ...
var InitFlags = klogv2.InitFlags
var Flush = klogv2.Flush

// ...
var Info = klogv2.Info
var Infoln = klogv2.Infoln
var InfoDepth = klogv2.InfoDepth
var Infof = klogv2.Infof

type Verbose bool

func V(level Level) Verbose {
	return Verbose(klogv2.V(level).Enabled())
}

func (v Verbose) Info(args ...interface{}) {
	if v {
		klogv2.Info(args)
	}
}

func (v Verbose) Infoln(args ...interface{}) {
	if v {
		klogv2.Infoln(args)
	}
}

func (v Verbose) Infof(format string, args ...interface{}) {
	if v {
		klogv2.Infof(format, args)
	}
}
```

**Comment by dims** [comment]
@44past4 this will help folks who are using `klog` in their main project but will not help them if they are vendoring some obscure dependency which still uses `klog` UNLESS they use say go mod replace commands to a repo with this code. (and folks use other tools like dep and vndr ... so i becomes harder!)

File: examples/coexist_klog_v1/coexist_klog_v1.go (hunk 4 above)
Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:11]: Fixed.

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	flag.Set("log_file", "myfile.log")
+	flag.Parse()
+
+	klog.Info("hello from klog v1!")
+	klogv2.Info("hello from klog v2!")
+	klogv2.Warning("beware from klog v2!")
+	klogv2.Error("error from klog v2!")
+	klog.Info("nice to meet you")
+	klog.Flush()
+}
```
Comment (dims) [inline examples/coexist_klog_v1/coexist_klog_v1.go:41]:
```
[dims@dims-a02 18:05] ~/go/src/k8s.io/klog/examples/coexist_klog_v1 ⟩ cat -p myfile.log
Log file created at: 2020/05/08 18:05:15
Running on machine: dims-a02
Binary: Built with gc go1.13.10 for darwin/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0508 18:05:15.825475   27901 coexist_klog_v1.go:35] hello from klog v1!
I0508 18:05:15.825772   27901 coexist_klog_v1.go:36] hello from klog v2!
W0508 18:05:15.826102   27901 coexist_klog_v1.go:37] beware from klog v2!
E0508 18:05:15.826417   27901 coexist_klog_v1.go:38] error from klog v2!
I0508 18:05:15.826421   27901 coexist_klog_v1.go:39] nice to meet you
```

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go (hunk 6 above)
Comment (dims) [reply examples/coexist_klog_v1_using_logr/coexist_klog_v1.go]: [dims@dims-a02 18:21] ~/go/src/k8s.io/klog/examples/coexist_klog_v1_using_logr ⟩ go run coexist_klog_v1.go && cat -p myfile.log && rm myfile.log
Log file created at: 2020/05/08 18:21:09
Running on machine: dims-a02
Binary: Built with gc go1.13.10 for darwin/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0508 18:21:09.815897   31798 coexist_klog_v1.go:47] hello from klog v1!
I0508 18:21:09.816189   31798 coexist_klog_v1.go:48] hello from klog v2!
I0508 18:21:09.816196   31798 coexist_klog_v1.go:49] beware from klog v2!
E0508 18:21:09.816202   31798 coexist_klog_v1.go:50] error from klog v2!
I0508 18:21:09.816205   31798 coexist_klog_v1.go:51] nice to meet you

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	flag.Set("stderrthreshold", "FATAL")
+	flag.Set("log_file", "myfile.log")
+	flag.Parse()
+
+	klog.Info("hello from klog v1!")
+	klogv2.Info("hello from klog v2!")
+	klogv2.Warning("beware from klog v2!")
+	klogv2.Error("error from klog v2!")
+	klog.Info("nice to meet you")
+	klog.Flush()
```
Comment (dims) [inline examples/coexist_klog_v1/coexist_klog_v1.go:40]: Output from this was:
```
Log file created at: 2020/05/08 18:23:23
Running on machine: dims-a02
Binary: Built with gc go1.13.10 for darwin/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0508 18:23:23.169339   31974 coexist_klog_v1.go:35] hello from klog v1!
I0508 18:23:23.169646   31974 coexist_klog_v1.go:36] hello from klog v2!
W0508 18:23:23.170001   31974 coexist_klog_v1.go:37] beware from klog v2!
E0508 18:23:23.170324   31974 coexist_klog_v1.go:38] error from klog v2!
I0508 18:23:23.170327   31974 coexist_klog_v1.go:39] nice to meet you
```

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+	flag.Set("stderrthreshold", "FATAL")
+	flag.Set("log_file", "myfile.log")
+	flag.Parse()
+
+	klog.Info("hello from klog v1!")
+	klogv2.Info("hello from klog v2!")
+	klogv2.Warning("beware from klog v2!")
+	klogv2.Error("error from klog v2!")
+	klog.Info("nice to meet you")
+	klog.Flush()
```
Comment (dims) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:52]: Output from here was:
```
Log file created at: 2020/05/08 18:24:24
Running on machine: dims-a02
Binary: Built with gc go1.13.10 for darwin/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0508 18:24:24.077871   32559 coexist_klog_v1.go:47] hello from klog v1!
I0508 18:24:24.078238   32559 coexist_klog_v1.go:48] hello from klog v2!
I0508 18:24:24.078244   32559 coexist_klog_v1.go:49] beware from klog v2!
E0508 18:24:24.078249   32559 coexist_klog_v1.go:50] error from klog v2!
I0508 18:24:24.078252   32559 coexist_klog_v1.go:51] nice to meet you
```

File: klog.go (hunk 1 above)
Comment (neolit123) [inline klog.go:834]: feels like a change in behavior. consumer using the current default (logtostderr = true) would not get it there anymore.
is it necessary for the examples of v1 + v2?

Comment (dims) [reply klog.go:834]: Yes, it's needed @neolit123

Comment (dims) [reply klog.go:834]: By default [toStdErrr is true](https://github.com/kubernetes/klog/blob/master/klog.go#L411). Since we will not call InitFlags, we end up skipping whatever we set up using SetOutput/SetOutputBySeverity because of this [piece of code](https://github.com/kubernetes/klog/blob/master/klog.go#L873-L875)

Comment (neolit123) [reply klog.go:834]: this is tricky...at minimum it needs a release note and even after that i'm not sure it should be done like so. are there workarounds without changing the user facing behavior, e.g. a new private bool var somewhere?

Comment (dims) [reply klog.go:834]: +1 to release note. if folks are NOT using those methods, they won't see any changes.

Comment (neolit123) [reply klog.go:834]: ok, understood. i'm going to defer my LGTM to an approver of klog.
e.g. @liggitt commented on the same problem earlier [here](https://github.com/kubernetes/klog/pull/142#discussion_r413056281).

Comment (liggitt) [reply klog.go:834]: https://github.com/kubernetes/klog/pull/100 proposed control over this variable...

I'm quite concerned that the existing methods have many unexpected side effects already, and people have worked around them in various ways, and that breaking those workarounds will cause significant pain downstream.

Comment (liggitt) [reply klog.go:834]: The current mix of flag-controlled aspects and programmatically controlled aspects is quite difficult to navigate. I don't think adding more side effects to the programmatic methods is what we want to do. Thinking as a consumer, the more 1:1 a setter is, the better.

In klog v1:

Log to stderr in all of the following cases:
* toStderr is true (only settable via flag, defaults to true)
* alsoToStderr is true (only settable via flag, defaults to false)
* go's standard log is being written to via something installed with CopyStandardLogTo(...)
* log message threshold is >= stderrthreshold (only settable via flag, defaults to error)

Log to file output **IFF** toStderr is false (only settable via flag, defaults to true) and at least one of the following is true:
  * `log_file` is set (only settable via flag, defaults to "")
  * SetOutput() was called
  * SetOutputBySeverity() was called and matches the message's severity

klog/v2 makes this more complicated by adding the logr interface as a third case, which takes precedence over all flags and outputs, as far as I can tell.

Comment (dims) [reply klog.go:834]: gotcha, let's merge https://github.com/kubernetes/klog/pull/100 and i'll remove the hacks here

Comment (dims) [reply klog.go]: @liggitt DONE! merged #100 and updated the examples to use that new method to disable toStdErr

**Comment by dims** [comment]
@serathius @liggitt @neolit123 @DirectXMan12 - 2 examples ready to merge!

File: examples/go.mod [hunk 10]
```diff
@@ -6,3 +6,5 @@ require (
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
 	k8s.io/klog/v2 v2.0.0-20200324194303-db919253a3bc
 )
+
+replace k8s.io/klog/v2 => ./..
```
Comment (liggitt) [inline examples/go.mod:10]: people will copy this and it will be incorrect for them (`./..` won't point to klog)

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/go-logr/logr"
+
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+type klogLogger struct{}
+
+func (kw klogLogger) Info(msg string, keysAndValues ...interface{}) {
+	klog.InfoDepth(5, msg)
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:14]: where does 5 come from? if this is based on internal routing within klog, we shouldn't recommend this (it will break on klog version updates)

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+func (kw klogLogger) Enabled() bool {
+	return true
+}
+
+func (kw klogLogger) Error(err error, msg string, keysAndValues ...interface{}) {
+	klog.ErrorDepth(5, msg)
+}
+
+func (kw klogLogger) V(level int) logr.InfoLogger {
+	return kw
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:26]: this drops verbosity info, right?

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func (kw klogLogger) WithValues(keysAndValues ...interface{}) logr.Logger {
+	return kw
+}
+
+func (kw klogLogger) WithName(name string) logr.Logger {
+	return kw
+}
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:35]: I'm confused about why discarding the data is ok... isn't the goal of this example is to show how klog/v2 and klog/v1 can coexist and work properly? This seems like it routes log data to the same destination at the price of dropping the data or changing verbosity behavior...

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+func main() {
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:37]: I know this is largely an artifact of the klog flag structure, but it would be really helpful to comment about why each step is required and what it is doing... as is, this is really hard to follow

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"flag"
+
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
```
Comment (liggitt) [inline examples/coexist_klog_v1/coexist_klog_v1.go:12]: this is clever, but I have a few questions:
* is it valid to assume every Write contains an entire message, where p[0] is the log header, or is Write ever called with partial lines or non-log-header lines?
* same question as https://github.com/kubernetes/klog/pull/142/files#r423834783 about depth of 6

Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: @liggitt yes. by the time it gets here it has the standard prefix for each line

yes the magic depth of 6 gets us the right file names. I don't see any other way to do this :(

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go (hunk 11 above)
Comment (dims) [reply examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:14]: same as above. I can add tests for both to fix the depth. will that help?

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go (hunk 12 above)
Comment (dims) [reply examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:26]: yea, i like the SetOutput solution better at this point

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go (hunk 13 above)
Comment (dims) [reply examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:35]: Yep same as above. the SetOutput one came out better than this one.

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go (hunk 14 above)
Comment (dims) [reply examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:37]: Agree. I will do that

File: examples/go.mod (hunk 10 above)
Comment (dims) [reply examples/go.mod:10]: Should i add a trailing comment? (that gets preserved right?)

File: examples/coexist_klog_v1/coexist_klog_v1.go (hunk 15 above)
Comment (liggitt) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: > is Write ever called with partial lines or non-log-header lines?

For example, logging a multi-line message, or a message longer than the default buffer size?

This turns the way klog/v2 makes `Write()` calls into an API... which is clever, but I could easily see causing problems in the future

Comment (liggitt) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: > the magic depth of 6 gets us the right file names. I don't see any other way to do this :(

do all calls to Write() occur at a constant distance from non-klog packages?

Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: yes they do

Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: Adding a test

Comment (liggitt) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: should we expose package constants (e.g. `OutputCallDepth`, `LoggerCallDepth`) that can be used for this? we could verify with tests they are correct, and if the klog internals change, we could adjust them as needed

Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: Ack will do that

Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:12]: Done!

File: examples/go.mod (hunk 10 above)
Comment (dims) [reply examples/go.mod:10]: added trailing comment

File: klog_overrides_test.go [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+import (
+	"testing"
+)
+
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
+	// 6 is the depth in the stack where the call to klog methods
+	// originate from. 53 is the length of the prefix before the
```
Comment (liggitt) [inline klog_overrides_test.go:23]: s/6/OutputCallDepth/
s/53/DefaultPrefixLength/

File: klog_overrides_test.go [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+	} else if p[0] == 'F' {
+		FatalDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	}
+	return len(p), nil
+}
+
+// TestInfoWithSetOutputBySeverity helps ensure that we do not alter some of the assumptions
+// used in the co-existence examples. Namely the depth of the stack where the klog v2 call
+// originates from and the length of the prefix. So folks who use SetOutput are guaranteed
+// what they see (depth = 6 and prefix length = 53, see Write method above)
```
Comment (liggitt) [inline klog_overrides_test.go:40]: don't reference specific numeric values here... just the const names

File: klog_overrides_test.go (hunk 16 above)
Comment (dims) [reply klog_overrides_test.go:23]: Done

File: klog_overrides_test.go (hunk 17 above)
Comment (dims) [reply klog_overrides_test.go:40]: Done

File: klog.go [hunk 18]
```diff
@@ -827,6 +831,8 @@ func SetLogger(logr logr.Logger) {
 	logging.logr = logr
 }
 
+const OutputCallDepth = 6
```
Comment (liggitt) [inline klog.go:834]: needs godoc explaining what this is useful for

File: klog.go [hunk 19]
```diff
@@ -815,6 +817,8 @@ func (rb *redirectBuffer) Write(bytes []byte) (n int, err error) {
 	return rb.w.Write(bytes)
 }
 
+const LoggerCallDepth = 5
```
Comment (liggitt) [inline klog.go:820]: needs godoc explaining what this is useful for

File: examples/coexist_klog_v1_using_logr/go.mod [hunk 20]
```diff
@@ -0,0 +1,10 @@
+module k8s.io/klog/examples/coexist_klog_v1
+
+go 1.13
+
+require (
+	k8s.io/klog v1.0.0
+	k8s.io/klog/v2 v2.0.0
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/go.mod:7]: v2.0.0 won't have the depth constants, will need to tag a v2.0.1 or something (applies everywhere)

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+	} else if p[0] == 'F' {
+		klog.FatalDepth(klogv2.OutputCallDepth, string(p[klogv2.DefaultPrefixLength:]))
+	}
+	return len(p), nil
+}
+
+func main() {
+	klog.InitFlags(nil)
+	klogv2.LogToStderr(false)
+	klogv2.SetOutputBySeverity("INFO", klogWriter{})
```
Comment (liggitt) [inline examples/coexist_klog_v1/coexist_klog_v1.go:28]: only these two lines are what they would need to add to their existing klog setup that is working the way they want in order to wire klog/v2 -> klog/v1, right?

should probably make it clear the `klog.InitFlags(nil)` and all the flag.Set stuff in this example only affects klog v1 and can be ignored if they already have klog v1 working as they want

File: examples/coexist_klog_v1_using_logr/coexist_klog_v1.go [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+
+func (kw klogLogger) WithName(name string) logr.Logger {
+	// if you start using structured logging, please serialize as well
+	return kw
+}
+
+func main() {
+	klog.InitFlags(nil)
+	klogv2.LogToStderr(false)
+	klogv2.SetLogger(klogLogger{})
```
Comment (liggitt) [inline examples/coexist_klog_v1_using_logr/coexist_klog_v1.go:45]: similar question... only the klogv2 lines are relevant for someone who already has klog v1 set up as desired?

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+	"flag"
+
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
+	if p[0] == 'I' {
```
Comment (liggitt) [inline examples/coexist_klog_v1/coexist_klog_v1.go:13]: add an initial `if len(p) < klogv2.DefaultPrefixLength { klog.InfoDepth(...) }` case?

File: examples/coexist_klog_v1/coexist_klog_v1.go (hunk 21 above)
Comment (liggitt) [reply examples/coexist_klog_v1/coexist_klog_v1.go:28]: also note that skipHeaders means the output writer doesn't get severity prefix

File: examples/coexist_klog_v1/coexist_klog_v1.go (hunk 23 above)
Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:13]: Done

File: examples/coexist_klog_v1/coexist_klog_v1.go [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+		klog.InfoDepth(klogv2.OutputCallDepth, string(p[klogv2.DefaultPrefixLength:]))
+	} else if p[0] == 'W' {
+		klog.WarningDepth(klogv2.OutputCallDepth, string(p[klogv2.DefaultPrefixLength:]))
+	} else if p[0] == 'E' {
+		klog.ErrorDepth(klogv2.OutputCallDepth, string(p[klogv2.DefaultPrefixLength:]))
+	} else if p[0] == 'F' {
+		klog.FatalDepth(klogv2.OutputCallDepth, string(p[klogv2.DefaultPrefixLength:]))
+	}
```
[the remark this answers is not in the document]
Comment (dims) [reply examples/coexist_klog_v1/coexist_klog_v1.go:21]: Done

**Comment by dims** [comment]
Served its purpose will open an example against release-1.x
